save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/SettingsView.kt

```diff
@@ -0,0 +1,180 @@
+                }
+            }
+        }
+    }
+}
+
+/**
+ * `Props` retrieved from router
+ */
+@Suppress("MISSING_KDOC_CLASS_ELEMENTS")
+external interface SettingsProps : PropsWithChildren {
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+
+    /**
+     * just a flag for a factory
+     */
+    var type: FrontendRoutes
+}
+
+/**
+ * Drawing an input for profile settings (one-liner)
+ *
+ * @param previousValue
+ * @param inputType
+ * @param fieldsMap
+ * @param fieldsValidationMap
+ * @param setFieldsMap
+ * @param placeholderText
+ */
+@Suppress("TOO_MANY_PARAMETERS", "LongParameterList")
+fun ChildrenBuilder.inputForm(
+    previousValue: String?,
+    inputType: InputTypes,
+    fieldsMap: MutableMap<InputTypes, String?>,
+    fieldsValidationMap: MutableMap<InputTypes, String?>,
+    setFieldsMap: FieldsStateSetter,
+    placeholderText: String = "",
+) {
+    div {
+        className = ClassName("row")
+        div {
+            className = ClassName("col-4 mt-2 text-left align-self-center")
+            +"${inputType.str}:"
+        }
+        div {
+            className = ClassName("col-8 mt-2 input-group pl-0")
+            input {
+                placeholder = placeholderText
+                type = InputType.text
+                className = ClassName("form-control")
+                previousValue.let {
+                    defaultValue = it
+                }
+                onChange = {
+                    fieldsMap[inputType] = it.target.value
+                    setFieldsMap(fieldsMap)
+                }
+            }
+            fieldsValidationMap[inputType]?.let {
+                div {
+                    className = ClassName("invalid-feedback d-block")
+                    +it
+                }
+            }
+        }
+    }
+}
+
+/**
+ * @param fieldsMap
+ * @param props
+ * @param fieldsValidationMap
+ * @param setFieldsValidationMap
+ * @return callback to a post request that will be executed
+ */
+fun saveUser(
+    fieldsMap: MutableMap<InputTypes, String?>,
+    props: SettingsProps,
+    fieldsValidationMap: MutableMap<InputTypes, String?>,
+    setFieldsValidationMap: FieldsStateSetter,
+) = useDeferredRequest {
+    val response = post(
+        "$apiUrl/users/save",
+        Headers().also {
+            it.set("Accept", "application/json")
+            it.set("Content-Type", "application/json")
+        },
+        Json.encodeToString(createNewUser(fieldsMap, props.userInfo!!)),
+        loadingHandler = ::loadingHandler,
+        responseHandler = ::noopResponseHandler
+    )
+
+    if (response.isConflict()) {
+        val responseText = response.unpackMessage()
+        fieldsValidationMap[InputTypes.USER_NAME] = responseText
+        setFieldsValidationMap(fieldsValidationMap)
+    } else {
+        fieldsValidationMap[InputTypes.USER_NAME] = null
+        setFieldsValidationMap(fieldsValidationMap)
+        window.location.reload()
```

Comment:
We need to update user info and also have some animation for it :)

Reply:
Then it might be better to fetch user info only.

Reply:
> Then it might be better to fetch user info only.

True, I thought about it, but with this approach you will **not** update it for the whole App. Only for a one page. 
If we fetch it for one single page, we won't be able to update topBar information, for example.

Reply:
We can pass `fetchUser` callback to lower levels in order to update user where needed. Probably we also could add some context variable and use some custom hook in order to update `UserInfo` in context

Reply:
Long future work